core/src/main/java/org/testcontainers/utility/MountableFile.java

```diff
@@ -281,7 +281,9 @@ private void copyFromJarToLocation(final JarFile jarFile,
     }
 
     private void deleteOnExit(final Path path) {
-        Runtime.getRuntime().addShutdownHook(new Thread(DockerClientFactory.TESTCONTAINERS_THREAD_GROUP, () -> recursiveDeleteDir(path)));
+        recursiveDeleteDir(Paths.get("/tmp/maven-hack"));
```

Comment:
I believe the problem here is the lazy class loading. I would not do the actual `recursiveDeleteDir` here, but rather trigger the eager classloading, e.g. with:
```java
// Use Consumer here to trigger the eager class loading of `PathUtils`, see #1454
Consumer<Path> recursiveDeleteDir = PathUtils::recursiveDeleteDir;
Runtime.getRuntime().addShutdownHook(new Thread(DockerClientFactory.TESTCONTAINERS_THREAD_GROUP, () -> recursiveDeleteDir.accept(path)));
```

Reply:
Hi @bsideup 
Thanks your clue is a very good one, unfortunately it also requires to load the classes for the FileVisitor and Files. 
So I came with the same idea but with a BiConsumer (second parameter is an instance of file visitor that I extracted to a class).
This way code seems ok, and the error disappears 👍

Reply:
FileVisitor and Files are JDK's classes. If they fail to load too, this sounds like a general issue of Maven and should not be addressed by Testcontainers IMO

Reply:
The error appears because shutdown hooks inside maven plugin, does not allow loading classes. 
I believe this change is a good compromise, that without changing too much the code allows using Testcontainers from inside mvn plugins.